pymc_marketing/clv/utils.py

```diff
@@ -597,3 +598,185 @@ def rfm_train_test_split(
+
+    for column_name in zip(
+        ["r_quartile", "f_quartile", "m_quartile"],
+        ["recency", "frequency", "monetary_value"],
+        strict=False,
+    ):
+
+        def _quartile_labels(column_name, max_label_range):
+            # recency labels must be reversed because lower values are more desirable
+            if column_name == "r_quartile":
+                return list(reversed(list(range(1, max_label_range))))
+            else:
+                return range(1, max_label_range)
+
+        try:
+            labels = _quartile_labels(column_name[0], 4)
+            rfm_data[column_name[0]] = pd.qcut(
+                rfm_data[column_name[1]], q=4, labels=labels, duplicates="drop"
+            ).astype(str)
+        except ValueError:
+            try:
+                labels = _quartile_labels(column_name[0], 3)
+                rfm_data[column_name[0]] = pd.qcut(
+                    rfm_data[column_name[1]], q=4, labels=labels, duplicates="drop"
+                ).astype(str)
+            except ValueError:
+                labels = _quartile_labels(column_name[0], 5)
+                rfm_data[column_name[0]] = pd.qcut(
+                    rfm_data[column_name[1]], q=4, labels=labels, duplicates="drop"
+                ).astype(str)
+
+    # TODO: Is `finally` block also needed?
+    rfm_data["rfm_score"] = (
+        rfm_data["r_quartile"] + rfm_data["f_quartile"] + rfm_data["m_quartile"]
+    )
+
+    if segment_config is None:
+        segment_config = {
+            "Premium Customer": [
+                "334",
+                "443",
+                "444",
+                "344",
+                "434",
+                "433",
+                "343",
+                "333",
+            ],
+            "Repeat Customer": ["244", "234", "232", "332", "143", "233", "243"],
+            "Top Spender": [
+                "424",
+                "414",
+                "144",
+                "314",
+                "324",
+                "124",
+                "224",
+                "423",
+                "413",
+                "133",
+                "323",
+                "313",
+                "134",
+            ],
+            "At Risk Customer": [
+                "422",
+                "223",
+                "212",
+                "122",
+                "222",
+                "132",
+                "322",
+                "312",
+                "412",
+                "123",
+                "214",
+            ],
+            "Inactive Customer": ["411", "111", "113", "114", "112", "211", "311"],
+        }
+
+    segment_names = list(segment_config.keys())
+
+    # create catch-all "Other" segment and assign defined segments from config
+    rfm_data["segment"] = "Other"
+
+    for key in segment_names:
+        rfm_data.loc[rfm_data["rfm_score"].isin(segment_config[key]), "segment"] = key
+
+    # drop unnecessary columns
+    # TODO: Is dropping the rfm_score column prudent?
```

Comment:
What about if we add this as a boolean option so that the user can choose?

Reply:
I'll get it added. Can also add columns for the bin edges.